### PR TITLE
Update dependencies and Dockerfile ROCm versions

### PR DESCRIPTION
**Pull request by junliume**
Some old dependencies have security risks and we are asked to update them if we can

### CODE REVIEW

File: src/solver/batchnorm/backward_ck.cpp [hunk 1]
```diff
@@ -160,7 +160,7 @@ bool BnCKBwdBackward::IsApplicable(
     [[maybe_unused]] const ExecutionContext& context,
     [[maybe_unused]] const miopen::batchnorm::ProblemDescription& bn_problem) const
 {
-#if MIOPEN_BACKEND_HIP || MIOPEN_USE_COMPOSABLEKERNEL
+#if MIOPEN_BACKEND_HIP && MIOPEN_USE_COMPOSABLEKERNEL
```
Comment (junliume) [inline src/solver/batchnorm/backward_ck.cpp]: @bghimireamd : please check this line and the two changed files. It seems that #2385 has broken `MIOPEN_USE_COMPOSABLEKERNEL=Off`

Comment (junliume) [reply src/solver/batchnorm/backward_ck.cpp]: I think these changes should belong to a separate PR, @bghimireamd @JehandadKhan could you create one?

Comment (bghimireamd) [reply src/solver/batchnorm/backward_ck.cpp]: good catch! I will create separate PR for this

Comment (bghimireamd) [reply src/solver/batchnorm/backward_ck.cpp]: @junliume PR created [#2464 ](https://github.com/ROCmSoftwarePlatform/MIOpen/pull/2464)

**Comment by junliume** [comment]
@atamazov @JehandadKhan @DrizztDoUrden @averinevg @CAHEK7 The PR is ready for review.

**Review by averinevg — APPROVED** [review]

**Review by atamazov — APPROVED** [review]
No objections provided that this works fine ;)

**Pull request merged**